Content.Server/Pinpointer/PinpointerSystem.cs

```diff
@@ -68,20 +72,34 @@ private void OnLocateTarget(ref FTLCompletedEvent ev)
         }
     }
 
-    private void LocateTarget(EntityUid uid, PinpointerComponent component)
+    /// <summary>
+    /// Searches the closest object that has a specific component for every component the pinpointer has stored.
+    /// This entity is then added to the stored targets.
+    /// </summary>
+    private void LocateTarget(EntityUid uid, PinpointerComponent component, EntityUid? user = null)
     {
         // try to find target from whitelist
-        if (component.IsActive && component.Component != null)
+        if (component.IsActive)
         {
-            if (!EntityManager.ComponentFactory.TryGetRegistration(component.Component, out var reg))
+            foreach (var preinstalledTarget in component.Components)
```

Comment:
You shouldn't be searching and storing every target every time you run locate target. Instead you should only search for the active target on the pinpointer, and allow the player select the type of target to look for on the list

Reply:
Locate target is only ran if the amount of stored targets is lower than the amount of pre-installed components when it's turned on and if FTL travel is completed, so it's not being called that often.
Currently in this PR there isn't even a pinpointer that tracks more than 1 component so it's only iterating once if the amount of stored targets is > the amount of components it has(they currently all have 1, namely: the nuke disk or the station).

Reply:
> Locate target is only ran in the following siuations: 1: If the amount of stored targets is lower than the amount of components given to a pinpointer when it's toggled from off to on . 2: If FTL travel is completed. So it's not really called that often.
> 
> Currently in this PR there isn't even a pinpointer that tracks more than 1 component so it's only iterating once if the amount of stored targets is > the amount of components it has(they currently all have 1, namely: the nuke disk or the station).
> 
> To summarize, it only gets called if there are 0 targets stored when the pinpointer is turned on, or when the pinpointer completes FTL travel. It currently also doesn't even do any loop because of each pinpointer only having 1 component, so if it's really a no-go I can remove the foreach without affecting the working state of this PR.

Reply:
> Instead you should only search for the active target on the pinpointer, and allow the player select the type of target to look for on the list

Do you mean it should have a seperate list in the verb menu for components(current list contains EntityId's not components) and call locate target every time a component is selected from that list?

Reply:
Ichanged it into what you requested